include/vectorSIMD.h

```diff
@@ -324,6 +324,12 @@ namespace core
 
 		inline vectorSIMD_32<T> operator-(T val) const { return (*this)-vectorSIMD_32<T>(val); }
 		inline vectorSIMD_32<T>& operator-=(T val) { return ( (*this) -= vectorSIMD_32<T>(val) ); }
+
+		inline vectorSIMD_32<T>& operator>>(uint32_t val) { _mm_store_si128(reinterpret_cast<__m128i*>(pointer), _mm_srai_epi32(Base::getAsRegister(),val)); return (*this); }

```

Comment:
shift right behaves differently on signed vs unsigned (whether it shifts in copies of the sign bit)

Reply:
I'm going to use it only for aabbSIMDu (unsigned int aabb) in getCenter() function, where i need to divide by 2.

Reply:
i.e. 
```cpp
(getMaxEdge() + getMinEdge) >> 2;
```

Reply:
that's fine(ish)

but you've just added a >> operator to a class that provides both `vectorSIMD32u` and `vectorSIMD32i` so you've introduced a bug

Reply:
_mm_srli_epi32 will do for unsigned int?

Reply:
correct